Integrate StraightLinePrograms code

lkastner opened the pull request:
This is to incorporate the `StraightLinePrograms` code into the `src` folder. I had to change a few exports and add an `Oscar.` here and there. The code got moved to `src` to unify the structure of the repo. I turned the `README.md` into a part of the docs. Since I don't really know what the code does, any feedback is appreciated.

lkastner committed:
StraightLinePrograms: Integrated into src/ and test/

lkastner committed:
StraightLinePrograms: Tweak exports to make tests run

lkastner committed:
StraightLinePrograms: Turn README.md into docs

benlorenz commented on docs/doc.main:
```diff
@@ -165,6 +165,10 @@
   ],
 
+  "StraightLinePrograms" => [
+      "intro.md",
```
```suggestion
      "StraightLinePrograms/intro.md",
```

lkastner replied:
Thanks!

benlorenz commented on src/StraightLinePrograms/StraightLinePrograms.jl:
```diff
@@ -4,7 +4,7 @@ import Base: +, -, *, ^, parent
 
```
```suggestion

import ..AbstractAlgebra: evaluate
```
this should avoid a conflict for the export

benlorenz committed:
Update src/StraightLinePrograms/StraightLinePrograms.jl

to fix 
WARNING: using StraightLinePrograms.evaluate in module Main conflicts with an existing identifier.
